.github/workflows/update-archives-list.yml

```diff
@@ -0,0 +1,36 @@
+name: update-archives-list
+run-name: Updating archives list
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * 0'
+jobs:
+  run-archive-script:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v4
+
+      - name: install python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: install pyvo and required dependencies
+        run: | 
+          python -m ensurepip --upgrade
+          pip install astropy
+          pip install requests
+          pip install pyvo
+
+      - name: run archives script
+        run: python fetch_archives.py
+
+      - run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add .
+          git commit -m "updated archives file"
+          git push
```

Comment:
Will the new revision of the tool be published to the toolshed if the action pushes directly to master?
PR could be better option anyway.

Also, I'm not sure, but as long as we change the integral part of the tool here, we need to bump its version, right?

Reply:
> The action could parse the help section and make a call for every urls and rewrite them if they are unavailable yes, I could make another script that does that and run it in the action or make another action specifically for that. Since it will modify the tool xml file maybe it would be better to have a specific action for that
> 
> If we don't want automatic push to toolshed when pushes are automatised we could discriminate against specific user inside the toolshed action right? Or yes it could be a PR rather than direct push and then it will be manually merged. Is that what you meant?

I also think PR is better at this time, although it may become a bit annoying to merge PRs every week, it's actually good to know what's being added at least at first.

> Regarding the version since it's only data files I don't think it would warrant a version bump but I'm not sure what that implies toolshed and usegalaxy wise.

I suppose tool data is part of the tool, so version change is warranted. But TBD.

> 
> I'm not sure of what happens if you push to toolshed without changing version, will the targeted version be overwritten?
> 
> Regarding galaxy.eu I think they pull the code directly from the toolshed when you access the site (I don't think they have all the tools locally) so if the usegalaxy tool version list already has the targeted version and the toolshed overwrite the code when changes are pushed to the same version it should be fine. But I'm really not sure about all that I will check if I can find more info

Is it possible that it pulls the version

Reply:
We agreed to increase the version.

Reply:
The question is still a case. The version needs to bumped in tool xml upon creating the PR.